custom_components/extron/media_player.py

```diff
@@ -11,26 +12,37 @@
 logger = logging.getLogger(__name__)
 
 
+def make_source_bidict(num_sources: int, input_names: list[str]) -> bidict:
+    bd = bidict()
+
+    for i in range(1, num_sources + 1):
+        # Use user-defined input name for the source when available
+        bd.put(i, input_names[i - 1] if i - 1 < len(input_names) else str(i))
+
+    return bd
```

Comment:
Whenever you find yourself first initializing something and then populating it one by one, there is typically a better way to do it in Python... Also swapped the indexing around since IMO it looks cleaner
```suggestion
    # Use user-defined input name for the source when available
    return bidict({i+1: input_names[i] if i < len(input_names) else str(i+1) for i in range(num_sources)})
```

Reply:
Good thing I wrote tests for this part, now we can easily change it :+1:

Reply:
Updated according to your suggestion, thanks!